### PR TITLE
[cli] surface bundling errors

### PR DESCRIPTION
**Pull request by alanjhughes**
# Why
Closes ENG-10223
Currently, if bundling fails, the app will start as normal but with an out of date bundle. 

# How
Added a new function that checks the build logs for any errors. If one is found, display to the user and throw a `CommandError`. Not sure if there is somewhere earlier I can do this and fail the build?

# Test Plan
Tested locally with nonexistent files and modules. Before the app would launch as normal, now you will see errors like these.

Module not found in node_modules
Screenshot 2023-09-27 at 15 20 26

Not found locally
Screenshot 2023-09-27 at 15 58 06

These errors are only output when building for release. They are the same as what will be presented in red box when running in dev.

### CODE REVIEW

**Comment by brentvatne** [comment]
@EvanBacon - do you think this is a good place for this logic, or should it be in [xcpretty](https://github.com/expo/third-party/tree/main/packages/xcpretty)?

File: packages/@expo/cli/src/run/ios/XcodeBuild.ts [hunk 1]
```diff
@@ -348,3 +349,18 @@ function getErrorLogFilePath(projectRoot: string): [string, string] {
...
+
+  // Unless we find an error, we don't need to do anything
+  if (lastErrorIndex === -1) {
+    return;
+  }
+
+  // Take 12 lines from the `Error:` line forwards
+  const block = lines.slice(lastErrorIndex, lastErrorIndex + 11).join('\n');
+
+  throw new CommandError(`Bundling Failed.\n${block}`);
```
Comment (brentvatne) [inline packages/@expo/cli/src/run/ios/XcodeBuild.ts:365]:
```suggestion
  throw new CommandError(`Bundling failed.\n${block}`);
```

**Review by EvanBacon — COMMENTED** [review]
What is the test case?

**Comment by alanjhughes** [comment]
@EvanBacon I've added some tests and a new file to the fixtures. To generate the error I followed the steps outlined in ENG-10223

**Comment by brentvatne** [comment]
@alanjhughes and i spoke about this and he plans to look into updating xcpretty so we'll get this same behavior on EAS Build

**Comment by alanjhughes** [comment]
Move [here](https://github.com/expo/third-party/pull/93)

**Pull request closed without merging**